src/Kentico.Xperience.Algolia/Indexing/DefaultAlgoliaClient.cs

```diff
@@ -0,0 +1,231 @@
+        (await searchClient.ListIndicesAsync(ct: cancellationToken).ConfigureAwait(false)).Items;
+
+    /// <inheritdoc/>
+    public async Task<ICollection<AlgoliaIndexStatisticsViewModel>> GetStatistics(CancellationToken cancellationToken) =>
+        (await GetStatisticsInternal(cancellationToken))
+        .Select(i => new AlgoliaIndexStatisticsViewModel
+        {
+            Name = i.Name,
+            Entries = i.Entries,
+            UpdatedAt = i.UpdatedAt
+        })
+        .ToList();
+
+    /// <inheritdoc />
+    public Task Rebuild(string indexName, CancellationToken cancellationToken)
+    {
+        if (string.IsNullOrEmpty(indexName))
+        {
+            throw new ArgumentNullException(nameof(indexName));
+        }
+
+        var algoliaIndex = AlgoliaIndexStore.Instance.GetIndex(indexName);
+        if (algoliaIndex == null)
+        {
+            throw new InvalidOperationException($"The index '{indexName}' is not registered.");
+        }
+
+        return RebuildInternal(algoliaIndex, cancellationToken);
+    }
+
+    /// <inheritdoc />
+    public Task<int> UpsertRecords(IEnumerable<JObject> dataObjects, string indexName, CancellationToken cancellationToken)
+    {
+        if (string.IsNullOrEmpty(indexName))
+        {
+            throw new ArgumentNullException(nameof(indexName));
+        }
+
+        if (dataObjects == null || !dataObjects.Any())
+        {
+            return Task.FromResult(0);
+        }
+
+        return UpsertRecordsInternal(dataObjects, indexName, cancellationToken);
+    }
+
+    private async Task<int> DeleteRecordsInternal(IEnumerable<string> objectIds, string indexName, CancellationToken cancellationToken)
+    {
+        var deletedCount = 0;
+        var searchIndex = await algoliaIndexService.InitializeIndex(indexName, cancellationToken);
+        var batchIndexingResponse = await searchIndex.DeleteObjectsAsync(objectIds, ct: cancellationToken).ConfigureAwait(false);
+        foreach (var response in batchIndexingResponse.Responses)
+        {
+            deletedCount += response.ObjectIDs.Count();
+        }
+
+        return deletedCount;
+    }
+
+    private async Task RebuildInternal(AlgoliaIndex algoliaIndex, CancellationToken cancellationToken)
+    {
+        var indexedItems = new List<IndexEventWebPageItemModel>();
+        foreach (var includedPathAttribute in algoliaIndex.IncludedPaths)
+        {
+            foreach (var language in algoliaIndex.LanguageNames)
+            {
+                var queryBuilder = new ContentItemQueryBuilder();
+
+                if (includedPathAttribute.ContentTypes != null && includedPathAttribute.ContentTypes.Count > 0)
+                {
+                    foreach (var contentType in includedPathAttribute.ContentTypes)
+                    {
+                        queryBuilder.ForContentType(contentType, config => config.WithLinkedItems(1).ForWebsite(algoliaIndex.WebSiteChannelName, includeUrlPath: true));
```

Comment:
```suggestion
                        queryBuilder.ForContentType(contentType, config => config.ForWebsite(algoliaIndex.WebSiteChannelName, includeUrlPath: true));
```

Remove `.WithLinkedItems(1)` because the developer will be using their index strategy to query for the data they need.

Reply:
removed